server/services/reviewService.ts

```diff
@@ -29,20 +30,23 @@ const toIncidentSummary = (namesByOffenderNumber: NamesByOffenderNumber) => (
   offenderNo: reportSummary.offenderNo,
 })
 
-// Surely this is defined somewhere useful? In a base set of typescript types perhaps?
-interface Predicate<T> {
-  (x: T): boolean
-}
+type Predicate<T> = (t: T) => boolean
 
 const offenderNameFilter = (nameToMatch?: string): Predicate<IncidentSummary> => {
-  const regexp = new RegExp(`^.*${nameToMatch}.*$`, 'i')
+  const safeString = nameToMatch.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
+  const regexp = new RegExp(`^.*${safeString}.*$`, 'i')
```

Comment:
Noticed that we were just accepting raw user provided string as regex - which would blow up if invalid regex passed and makes the service susceptible to regex bombs (by authenticated users)

Reply:
cool